### PR TITLE
Add problem matchers for pylint

### PR DESCRIPTION
**Pull request by nishipy**
## What changes are proposed in this pull request?

Add problem matchers for pylint.
Note that this PR includes a dummy code change now to confirm the behavior of problem matchers, so we have to roll back it.

## How is this patch tested?

Existing tests.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by nishipy** [comment]
It seems to work well.
Screen Shot 2022-04-08 at 22 30 12

File: .github/workflows/matchers/pylint.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          {
+            "regexp": "^(.+):(\\d+):(\\d+):\\s(([CRW]\\d{4}):\\s.+)$",
+            "file": 1,
+            "line": 2,
+            "column": 3,
+            "message": 4,
+            "code": 5
+          }
+        ]
+      }
```
Comment (harupy) [inline .github/workflows/matchers/pylint.json:30]:
```suggestion
      }
```

@nishipy Let's remove this :)

File: .github/workflows/matchers/pylint.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "problemMatcher": [
+      {
+        "owner": "pylint-error",
+        "severity": "error",
+        "pattern": [
+          {
+            "regexp": "^(.+):(\\d+):(\\d+):\\s(([EF]\\d{4}):\\s.+)$",
```
Comment (harupy) [inline .github/workflows/matchers/pylint.json:8]:
```suggestion
            "regexp": "^(.+):(\\d+):(\\d+):\\s(([A-Z]\\d{4}):\\s.+)$",
```

Can we fix the regular expression?

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -41,6 +41,9 @@ jobs:
       run: |
         source ./dev/install-common-deps.sh
         pip install -r requirements/lint-requirements.txt
+    - name: Add problem matchers
+      run: |
+        echo "::add-matcher::.github/workflows/matchers/pylint.json"
```
Comment (harupy) [inline .github/workflows/master.yml:46]: Can we move this step before the `install-dependencies` step?

File: .github/workflows/matchers/pylint.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+    "problemMatcher": [
+      {
+        "owner": "pylint-error",
```
Comment (harupy) [inline .github/workflows/matchers/pylint.json:4]:
```suggestion
        "owner": "pylint",
```

File: tests/helper_functions.py [hunk 5]
```diff
@@ -154,8 +154,7 @@ def pyfunc_serve_and_score_model(
                                              declaring the scoring process to have failed.
     :param extra_args: A list of extra arguments to pass to the pyfunc scoring server command. For
                        example, passing ``extra_args=["--env-manager", "local"]`` will pass the
-                       ``--env-manager local`` flag to the scoring server to ensure that conda
-                       environment activation is skipped.
+                       ``--env-manager local`` flag to the scoring server to ensure that conda environment activation is skipped.
     """
     env = dict(os.environ)
```
Comment (harupy) [inline tests/helper_functions.py:159]: Can we add a = 1 (unused variable) above this line as a test?

pylint should raise this error: http://pylint-messages.wikidot.com/messages:e0602

**Comment by harupy** [comment]
@nishipy Thanks for the PR, left some comments!

File: tests/helper_functions.py [hunk 6]
```diff
@@ -154,8 +154,7 @@ def pyfunc_serve_and_score_model(
                                              declaring the scoring process to have failed.
     :param extra_args: A list of extra arguments to pass to the pyfunc scoring server command. For
                        example, passing ``extra_args=["--env-manager", "local"]`` will pass the
-                       ``--env-manager local`` flag to the scoring server to ensure that conda
-                       environment activation is skipped.
+                       ``--env-manager local`` flag to the scoring server to ensure that conda environment activation is skipped.
```
Comment (harupy) [inline tests/helper_functions.py:157]: Nice! Let's fix `.github/workflows/matchers/pylint.json` first and make sure the warning will turn into an error.

**Comment by nishipy** [comment]
Thanks for correcting! fixed them and added some dummy statements.

File: .github/workflows/matchers/pylint.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
+    "problemMatcher": [
```
Comment (harupy) [inline .github/workflows/matchers/pylint.json:2]:
```suggestion
  "problemMatcher": [
```

Can we use 2 spaces for indent?

Comment (harupy) [reply .github/workflows/matchers/pylint.json:2]: Can we apply the same fix to other lines?

Comment (nishipy) [reply .github/workflows/matchers/pylint.json:2]: Sorry. I will fix the indent.

**Comment by harupy** [comment]
Can we fix the DCO check? You can click the details link and follow the instructions there.

**Comment by harupy** [comment]
Let's revert the dummy changes!

**Comment by harupy** [comment]
Can you try this?

```
git reset $(git merge-base master $(git branch --show-current))
git add -A
git commit -m "<commit message>"
```

**Comment by nishipy** [comment]
Thank you! It is what I needed.

**Comment by nishipy** [comment]
It works well so let me revert them.
Screen Shot 2022-04-09 at 1 30 29

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by nishipy** [comment]
@harupy Thanks for your continued support.
Let me know if you will have another task that I could help.

**Comment by harupy** [comment]
Thanks @nishipy! I have one task in my mind, which is to introduce isort to sort import statements in pylint scripts. Would you be interested on working on this?

**Comment by nishipy** [comment]
Sure! I'd like to work on it.
Does the following example help us?
https://github.com/pydata/bottleneck/actions/runs/65607935/workflow#L61

**Comment by nishipy** [comment]
It could be useful for us to add isort for the pre-commit hook.

**Comment by harupy** [comment]
Hi @nishipy, we discussed whether we should introduce isort, and our decision is no because it  adds overhead / friction to the contribution process. Let me find a different task.

**Pull request merged**